07-routing-gossip.md

```diff
@@ -144,30 +144,50 @@ nodes for which a channel is not already known are ignored.
 2. data:
    * [64:signature]
    * [4:timestamp]
-   * [16:ipv6]
-   * [2:port]
    * [33:node-id]
    * [3:rgb-color]
    * [32:alias]
-   * [2:len]
-   * [len:features]
+   * [2:flen]
+   * [flen:features]
+   * [2:addrlen]
+   * [addrlen:addresses]
 
 The `timestamp` allows ordering in the case of multiple announcements;
-the `ipv6` and `port` allow the node to announce its willingness to
-accept incoming network connections, the `rgb-color` and `alias` allow
+the `rgb-color` and `alias` allow
 intelligence services to give their nodes cool monikers like IRATEMONK
 and WISTFULTOLL and use the color black.
 
+`addresses` allows the node to announce its willingness to accept
+incoming network connections: it contains series of `address
+descriptor`s for connecting to the node.  The first byte describes the
+address type, followed by the number of bytes for that type.
+
+The following `address descriptor` types are defined:
+
+1. `0`: padding.  data = none (length 0).
+1. `1`: IPv4. data = `[4:ipv4-addr][2:port]` (length 6)
+2. `2`: IPv6. data = `[16:ipv6-addr][2:port]` (length 18)
+
 ### Requirements
 
 The creating node MUST set `timestamp` to be greater than any previous
 `node_announcement` it has created.  It MAY base it on a UNIX
-timestamp.  It MUST set the `ipv6` and `port` fields to all zeroes, or
-a non-zero `port` and `ipv6` set to a valid IPv6 address or an IPv4-Mapped IPv6 Address format as defined in [RFC 4291 section 2.5.5.2](https://tools.ietf.org/html/rfc4291#section-2.5.5.2).  It MUST set `signature` to the signature of
+timestamp.  It MUST set `signature` to the signature of
 the double-SHA256 of the entire remaining packet after `signature` using the
 key given by `node-id`.  It MAY set `alias` and `rgb-color` to customize their node's appearance in maps and graphs, where the first byte of `rgb` is the red value, the second byte is the green value and the last byte is the blue value.  It MUST set `alias` to a valid UTF-8 string, with any `alias` bytes following equal to zero.
 
-The creating node SHOULD set `len` to the minimum length required to
+The creating node SHOULD fill `addresses` with an address descriptor
+for each distinct network address which expects incoming connections.
```

Comment:
I think it'd reasonable to limit this to 1 instance for each address class. First of all it's going to be by far the most common case that a node just announces one address, and secondly a node can still announce different addresses by announcing a different `node_id` associated with that address.